OgreMain/src/OgreRenderQueueSortingGrouping.cpp

```diff
@@ -269,11 +265,9 @@ namespace {
             // Hmm, a bit hacky but least obtrusive for now
                     OGRE_LOCK_MUTEX(Pass::msPassGraveyardMutex);
             const Pass::PassSet& graveyardList = Pass::getPassGraveyard();
-            Pass::PassSet::const_iterator gi, giend;
-            giend = graveyardList.end();
-            for (gi = graveyardList.begin(); gi != giend; ++gi)
+            for (const auto& gi : graveyardList)
```

Comment:
```suggestion
            for (auto* p : graveyardList)
```
and the instance below